src/mapping/device/CudaQRSolver.cuh

```diff
@@ -0,0 +1,11 @@
+#ifdef PRECICE_WITH_CUDA
+#pragma once
+
+#include <ginkgo/ginkgo.hpp>
+
+/**
+ * Computes the QR decomposition using CUDA
+*/
+void computeQRDecompositionCuda(const int deviceId, const std::shared_ptr<gko::Executor> &exec, gko::matrix::Dense<> *A_Q, gko::matrix::Dense<> *R);
```

Comment:
Should we rather use `std::shared_ptr<>`  (or a reference to it) instead of a `std::shared_ptr<>*`?